File: packages/javascript/lib/word/randPlace.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * Places — where people gather, live and pass through.
 *
 * `randWord({ theme: 'place' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randPlace({ language: 'ko', count: 3 }); // ['시장', '광장', '마을']
 * randPlace({ language: 'en', count: 3 }); // ['Market', 'Plaza', 'Village']
 */
export function randPlace(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Place words with the language and theme each one came from.
 *
 * @example
 * randPlace({ language: 'ko', output: 'detail' });
 * // [{ word: '시장', language: 'ko', theme: 'place' }]
 */
export function randPlace(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randPlace(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('place', options);
}

File: packages/dart/lib/src/word/rand_place.dart
import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// Places — where people gather, live and pass through.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.place] to `randWordDetails` — Dart has
/// no overloads, and seventeen more functions for it would be seventeen too many.
///
/// ```dart
/// randPlace(language: WordLanguage.ko, count: 3); // [시장, 광장, 마을]
/// randPlace(language: WordLanguage.en, count: 3); // [Market, Plaza, Village]
/// ```
List<String> randPlace({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,
}) => randWord(
  language: language,
  theme: WordTheme.place,
  count: count,
  realism: realism,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
);
